src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,90 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+import com.intellij.psi.tree.IElementType
+import com.intellij.psi.tree.TokenSet
+
+interface RustAlignmentStrategy {
+    /**
+     * Requests current strategy for alignment to use for given child.
+     */
+    fun getAlignment(child: ASTNode, parent: ASTNode?, childCtx: RustFmtContext): Alignment?
+
+    /**
+     * Always returns `null`.
+     */
+    object NullStrategy : RustAlignmentStrategy {
+        override fun getAlignment(child: ASTNode, parent: ASTNode?, childCtx: RustFmtContext): Alignment? = null
+    }
+
+    /**
+     * Apply this strategy only when child element is in [tt].
+     *
+     * @param isBlackList   denotes how filter set is interpreted. If set to `false`, the filter is a white list:
+     *                      only children which element types are in filter set will get wrapped alignment;
+     *                      otherwise, the filter works as a black list: these children will be ignored.
+     */
+    fun filter(vararg tt: IElementType, isBlackList: Boolean = false): RustAlignmentStrategy =
```

Comment:
`isBlackList` is never used. What do you think about renaming this to `alignIf`, and the adding `notAlignIf` in the future when `balklisting` will be needed?